Training/Program.cs

```diff
@@ -1,7 +1,107 @@
-﻿namespace Training {
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for new interns at Metamation, Batch - July 2023
+// Copyright (c) Metamation India.                                              
+// ------------------------------------------------------------------------
+// Implement double ended queue using array of T
+// This is like a queue but we can add or remove
+// i.e. enqueue or dequeue at both the ends at the head and the tail. 
+// This should allow to add at the head or remove at the head and
+// add to the tail or remove at the tail.
+// --------------------------------------------------------------------------------------------
+namespace Training {
    internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      static void Main () { }
+   }
+   #region Queue ------------------------------------------------------------------------------
+
+   public class Deque<T> {
+      #region Properties --------------------------------------------
+      /// <summary>Capacity of the TDeque</summary>
+      public int Capacity => mElements.Length;
+
+      /// <summary>Count of the elements in TDeque</summary>
+      public int Count => mCount;
+
+      /// <summary>Tells if the queue is empty or not</summary>
+      public bool IsEmpty => mCount == 0;
+      #endregion
+
+      #region Method ------------------------------------------------
+      /// <summary>Throws exception when queue is empty</summary>
+      /// <exception cref="InvalidOperationException"></exception>
+      void CheckEmpty () {
+         if (IsEmpty) throw new InvalidOperationException ("Empty Queue");

```

Comment:
This is private method. So, summary comments are not required. Only public members should have summary like comments. 
```suggestion
      // Throws exception when queue is empty
      void CheckEmpty () {
         if (IsEmpty) throw new InvalidOperationException ("Empty Queue");

```

Reply:
Reason for including summary comments is to make sure if public members are used by external user they can see the documentation using xml file.  See [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#d5-an-example) for more details.